### PR TITLE
agent: backhaul_manager: fix remove_client_from_all_radios

### LINKED ISSUE #1241: [BUG] [UGW] monitor crash after client association
spotted on c960f8288198e4ce3b2caba73cf258b19b8547b2

steps to reproduce:

1. flash a version with latest prplmesh
2. configure SSID
3. print bml-conn-map**
4. connect a client 

expected behavior:
steps 1-4 are successful

actual behavior:
steps 1-3 are successful
step 4 results with a beerocks-agent crash
attached logs of /var/log/messaged (UGW) and beerocks logs

update:
the issue was introduced as part of the following PR - need to understand what change in the PR caused it:
https://github.com/prplfoundation/prplMesh/pull/888/commits

[prplmesh_logs.tar.gz](https://github.com/prplfoundation/prplMesh/files/4592143/prplmesh_logs.tar.gz)

### PR DESCRIPTION
**Pull request by arnout**
Fixes #1241.

Include a test.
It doesn't actually test the functionality of that function (we'd need to parse topology response for that), but it does trigger the crash from #1241.

### CODE REVIEW

File: tests/connmap.py [hunk 1]
```diff
@@ -14,12 +14,24 @@
 RE_MAC = rb"(?P<mac>([0-9a-fA-F]{2}:){5}[0-9a-fA-F]{2})"
 
 
+class MapClient:
+    '''Represents a client (STA) in the connection map.'''
+    def __init__(self, mac: str):
+        self.mac = mac
+
+
```
Comment (tomereli) [inline tests/connmap.py:22]: Extra line break - doesn't flake8 complain about this?

**Review by tomereli — APPROVED** [review]

**Comment by tomereli** [comment]
@itayx please confirm that this resolves the issue.

**Comment by itayx** [comment]
> @itayx please confirm that this resolves the issue.

This fixes the client association described in #1241 
In addition I added a fix that was discussed in the Slack #prplmesh-dev
https://prplfoundation.slack.com/archives/CKKLX17P1/p1588845700382900

**Review by itayx — APPROVED** [review]

**Review by yaniv-yitzhak — APPROVED** [review]

File: tests/connmap.py (hunk 1 above)
Comment (arnout) [reply tests/connmap.py:22]: I don't see the extra line break... PEP8 specifies two empty lines before each top-level class or function definition.

**Comment by arnout** [comment]
Itay's additional fix LGTM, so merging.

**Pull request merged**